commands/system/define-word.swift

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/swift
+
+// Required parameters:
+// @raycast.schemaVersion 1
+// @raycast.title Define Word
+// @raycast.author Jesse Claven
+// @raycast.authorURL https://github.com/jesse-c
+// @raycast.description Define a word using the built-in dictionary/dicionaries.
+// @raycast.packageName System
+// @raycast.mode fullOutput
+//
+// Optional parameters:
+// @raycast.icon 🗣
+// @raycast.argument1 { "type": "text", "placeholder": "Word (e.g. isthmus)" }
+
+import Foundation
+
+if CommandLine.argc != 2 {
+  print("Must pass 1 word to define")
+} else {
+  let argument = CommandLine.arguments[1].lowercased()
+
+  // https://developer.apple.com/library/archive/documentation/UserExperience/Conceptual/DictionaryServicesProgGuide/access/access.html#//apple_ref/doc/uid/TP40006152-CH5-SW3
+  let result = DCSCopyTextDefinition(nil, argument as CFString, CFRangeMake(0, argument.count))?.takeRetainedValue() as String?
+
+  print(result ?? "No definition found")
+}
```

Comment:
I suggest you to check if the amount of arguments if you received is greater than 1, then you will ensure that you can ask for the argument in the second position (`[1]`) without receive an `index out of bound` error:

```suggestion
if CommandLine.argc > 1 {
  let argument = CommandLine.arguments[1].lowercased()

  // https://developer.apple.com/library/archive/documentation/UserExperience/Conceptual/DictionaryServicesProgGuide/access/access.html#//apple_ref/doc/uid/TP40006152-CH5-SW3
  let result = DCSCopyTextDefinition(nil, argument as CFString, CFRangeMake(0, argument.count))?.takeRetainedValue() as String?

  print(result ?? "No definition found")
} else {
  print("Must pass 1 word to define") 
}
```

Reply:
That's a fair suggestion! Thanks.